bin/i18n/redact_restore_utils.rb

```diff
@@ -57,70 +68,54 @@ def self.restore_data(source_data, redacted_data, plugins = [], format = 'md')
+    if I18nScriptUtils.json_file?(source) || I18nScriptUtils.yaml_file?(source)
+      source_data = I18nScriptUtils.parse_file(source)
+      redacted_data = I18nScriptUtils.parse_file(redacted)
 
-    restored =
-      if is_json
-        RedactRestoreUtils.restore_data(source_data, redacted_data, plugins, format)
-      else
-        RedactRestoreUtils.restore_data(source_data.values.first, redacted_data.values.first, plugins, format)
-      end
+      return unless source_data
+      return unless redacted_data
+      return unless source_data&.values&.first&.length
+      return unless redacted_data&.values&.first&.length
 
-    File.open(dest, "w+") do |file|
-      if File.extname(dest) == '.json'
-        file.write(JSON.pretty_generate(restored))
-      else
+      # yaml files have a locale key wrapping all content. Redacted files have the translated language key, while the
+      # un-redacted file keeps the English key. We need to extract the data inside the locale key to restore content.
+      if I18nScriptUtils.yaml_file?(source)
+        restored = RedactRestoreUtils.restore_data(source_data.values.first, redacted_data.values.first, plugins, format)
         redacted_key = redacted_data.keys.first
         restored = {redacted_key => restored}
-        file.write(I18nScriptUtils.to_crowdin_yaml(restored))
+        I18nScriptUtils.write_yaml_file(dest, restored)
+      else
+        restored = RedactRestoreUtils.restore_data(source_data, redacted_data, plugins, format)
+        I18nScriptUtils.write_json_file(dest, restored)
       end
-    end
 
-    restored
+    else
+      restored = RedactRestoreUtils.restore_file(source, redacted, plugins, format)
+      I18nScriptUtils.write_file(dest, restored)
+    end
   end
 
+  # redact redacts the content of the source file, whether is a json, yml or other formats and write the output
+  # into the dest file.
   def self.redact(source, dest, plugins = [], format = 'md')
     return unless File.exist? source
-    FileUtils.mkdir_p File.dirname(dest)
 
-    source_data =
-      if File.extname(source) == '.json'
-        JSON.load_file(source)
-      else
-        YAML.load_file(source)
-      end
-
-    redacted = RedactRestoreUtils.redact_data(source_data, plugins, format)
+    if I18nScriptUtils.json_file?(source) || I18nScriptUtils.yaml_file?(source)
+      source_data = I18nScriptUtils.parse_file(source)
+      redacted = RedactRestoreUtils.redact_data(source_data, plugins, format)
 
-    File.open(dest, "w+") do |file|
-      if File.extname(dest) == '.json'
-        file.write(JSON.pretty_generate(redacted))
-      else
-        file.write(I18nScriptUtils.to_crowdin_yaml(redacted))
-      end
+      I18nScriptUtils.write_yaml_file(dest, redacted) if I18nScriptUtils.yaml_file?(source)
+      I18nScriptUtils.write_json_file(dest, redacted) if I18nScriptUtils.json_file?(source)
+    else
+      redacted = redact_file(source_data, plugins, format)
```

Comment:
Should this just be
```suggestion
      redacted = redact_file(source, plugins, format)
```
?

Reply:
I agree.